src/test/java/org/mule/test/http/functional/listener/HttpBackPressureTestCase.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.http.functional.listener;
+
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.DROPPED;
+import static org.mule.runtime.http.api.HttpConstants.HttpStatus.SERVICE_UNAVAILABLE;
+import static org.mule.tck.probe.PollingProber.check;
+import org.mule.runtime.core.api.util.func.CheckedRunnable;
+import org.mule.runtime.http.api.HttpConstants.HttpStatus;
+import org.mule.runtime.http.api.HttpService;
+import org.mule.runtime.http.api.client.HttpClient;
+import org.mule.runtime.http.api.client.HttpClientConfiguration;
+import org.mule.runtime.http.api.domain.entity.ByteArrayHttpEntity;
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.tck.junit4.rule.DynamicPort;
+import org.mule.test.http.functional.AbstractHttpTestCase;
+
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.inject.Inject;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class HttpBackPressureTestCase extends AbstractHttpTestCase {
+
+  @Rule
+  public DynamicPort listenPort = new DynamicPort("port");
+
+  @Inject
+  private HttpService httpService;
```

Comment:
Why was this necessary? `AbstractHttpTestCase` has a client available.

Reply:
simply because I didn't know that